elk/training/ccs_reporter.py

```diff
@@ -262,16 +267,19 @@ def fit(self, hiddens: Tensor) -> float:
         n, v, _ = x_neg.shape
         prompt_ids = torch.eye(v, device=x_neg.device).expand(n, -1, -1)
 
-        self.norm.update(
-            x=x_neg,
-            # Independent indicator for each (template, pseudo-label) pair
-            y=torch.cat([torch.zeros_like(prompt_ids), prompt_ids], dim=-1),
-        )
-        self.norm.update(
-            x=x_pos,
-            # Independent indicator for each (template, pseudo-label) pair
-            y=torch.cat([prompt_ids, torch.zeros_like(prompt_ids)], dim=-1),
-        )
+        if self.config.norm == "leace":
+            # type ignore because otherwise throws error, probably bug with pyright
```

Comment:
It's not a bug with Pyright, it's due to the fact that the type checker doesn't have a way to reliably determine the class of `self.norm` here. If you do an `isinstance` check on `self.norm` it would work.

Reply:
nice!